pipeline.cpp

```diff
@@ -81,6 +92,38 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
     // Your code goes here
+    // Code adapted from www.sunshine2k.de/coding/java/TriangleRasterization/TriangleRasterization.html
+    // This is known as the BaryCentric Algorithm. 
+    // Find the bounding box of the triangle, test each point in the box to see if it is in the triangle, if so, draw that pixel.
+    int maxX = fmax(triangle[0].x, fmax(triangle[1].x, triangle[2].x)); // Find the max x vertex by searching through the verticies of the triangle
```

Comment:
Minor: He gave us a global definition called MIN3 and MAX3 which takes the three points in question. Could be a little faster than importing the library but at the end of the day this is perfectly fine for this project. There could be other useful definitions in the definitions.h file for later weeks as well.